projects/opendr_ws_2/src/opendr_perception/opendr_perception/object_detection_2d_detr_node.py

```diff
@@ -47,64 +52,168 @@ def __init__(self, input_rgb_image_topic="image_raw", output_rgb_image_topic="/o
+            self.detection_publisher = self.create_publisher(Detection2DArray, detections_topic, 10)
         else:
-            self.object_publisher = None
+            self.detection_publisher = None
 
-        self.bridge = ROS2Bridge()
+        self.image_subscriber = self.create_subscription(ROS_Image, input_rgb_image_topic, self.callback, 10)
 
-        self.object_detector = DetrLearner(device=device)
-        self.object_detector.download(path=".", verbose=True)
+        self.bridge = ROS2Bridge()
 
-        self.get_logger().info("Object Detection 2D DETR node initialized.")
+        self.class_names = [
+            "N/A",
+            "person",
+            "bicycle",
+            "car",
+            "motorcycle",
+            "airplane",
+            "bus",
+            "train",
+            "truck",
+            "boat",
+            "traffic light",
+            "fire hydrant",
+            "N/A",
+            "stop sign",
+            "parking meter",
+            "bench",
+            "bird",
+            "cat",
+            "dog",
+            "horse",
+            "sheep",
+            "cow",
+            "elephant",
+            "bear",
+            "zebra",
+            "giraffe",
+            "N/A",
+            "backpack",
+            "umbrella",
+            "N/A",
+            "N/A",
+            "handbag",
+            "tie",
+            "suitcase",
+            "frisbee",
+            "skis",
+            "snowboard",
+            "sports ball",
+            "kite",
+            "baseball bat",
+            "baseball glove",
+            "skateboard",
+            "surfboard",
+            "tennis racket",
+            "bottle",
+            "N/A",
+            "wine glass",
+            "cup",
+            "fork",
+            "knife",
+            "spoon",
+            "bowl",
+            "banana",
+            "apple",
+            "sandwich",
+            "orange",
+            "broccoli",
+            "carrot",
+            "hot dog",
+            "pizza",
+            "donut",
+            "cake",
+            "chair",
+            "couch",
+            "potted plant",
+            "bed",
+            "N/A",
+            "dining table",
+            "N/A",
+            "N/A",
+            "toilet",
+            "N/A",
+            "tv",
+            "laptop",
+            "mouse",
+            "remote",
+            "keyboard",
+            "cell phone",
+            "microwave",
+            "oven",
+            "toaster",
+            "sink",
+            "refrigerator",
+            "N/A",
+            "book",
+            "clock",
+            "vase",
+            "scissors",
+            "teddy bear",
+            "hair drier",
+            "toothbrush",
+        ]
+
+        # Initialize the detection estimation
+        self.detr_learner = DetrLearner(device=device)
+        self.detr_learner.download(path=".", verbose=True)
```

Comment:
Learner should be renamed to `object_detector` to match the rest of the object detectors 2D.

```suggestion
        self.object_detector = DetrLearner(device=device)
        self.object_detector.download(path=".", verbose=True)
```